src/app/pages/dashboard/corona-map/corona-map.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, OnDestroy } from '@angular/core';
+import { NbThemeService } from '@nebular/theme';
+import { HttpClient } from '@angular/common/http';
+import { combineLatest } from 'rxjs';
+import { takeWhile } from 'rxjs/operators';
+import { registerMap } from 'echarts';
+
+@Component({
+  selector: 'ngx-corona-map',
+  templateUrl: './corona-map.component.html',
+  styleUrls: ['./corona-map.component.scss'],
+})
+export class CoronaMapComponent implements OnDestroy {
+  options: any;
+
+  private alive = true;
+
+  constructor(private theme: NbThemeService, private http: HttpClient) {
+    combineLatest([
+      this.http.get('assets/map/world.json'),
+      this.http.get('assets/map/israel.json'),
+      this.theme.getJsTheme(),
+    ])
+      .pipe(takeWhile(() => this.alive))
```

Comment:
use `takeUntil`and a subject.
I don't think this will work correctly

Reply:
It is not my code and it will work